csrc/util.h

```diff
@@ -42,96 +44,94 @@ inline std::string opensslErrorWithDefault(const char *fallback) {
 #define EX_ILLEGAL_ARGUMENT "java/lang/IllegalArgumentException"
 #define EX_ILLEGAL_STATE "java/lang/IllegalStateException"
 #define EX_INVALID_KEY "java/security/InvalidKeyException"
+#define EX_INVALID_KEY_SPEC "java/security/spec/InvalidKeySpecException"
 #define EX_SIGNATURE_EXCEPTION "java/security/SignatureException"
 
-// Define this prior to use as some compilers don't like it the other way around.
-static inline void secureZero(void *ptr, size_t size) {
-    if (ptr == nullptr || size == 0) {
+  // Define this prior to use as some compilers don't like it the other way around.
+  static inline void secureZero(void *ptr, size_t size)
+  {
+    if (ptr == nullptr || size == 0)
+    {
       return;
     }
     memset(ptr, 0, size);
     __asm__ __volatile__(
-        "" /* don't actually do anything */
-        :  /* no outputs */
-        : "r" (ptr) /* make the compiler think the memset matters */
-        : "memory"  /* pretend we modify memory, so the compiler can't cache values in registers */
+        ""         /* don't actually do anything */
+        :          /* no outputs */
+        : "r"(ptr) /* make the compiler think the memset matters */
+        : "memory" /* pretend we modify memory, so the compiler can't cache values in registers */
     );
-}
+  }
 
-template<typename type, size_t size>
-class SecureBuffer {
+  template <typename type, size_t size>
+  class SecureBuffer
+  {
   public:
     type buf[size];
 
     SecureBuffer() { secureZero(buf, sizeof(buf)); }
     virtual ~SecureBuffer() { zeroize(); }
-    operator type*() {return buf; }
-    operator const type*() const { return buf; }
-    type& operator*() { return &buf; }
-    const type& operator*() const { return &buf; }
-    type& operator[](size_t idx) { return buf[idx]; }
-    type& operator[](size_t idx) const { return buf[idx]; }
+    operator type *() { return buf; }
+    operator const type *() const { return buf; }
+    type &operator*() { return &buf; }
+    const type &operator*() const { return &buf; }
+    type &operator[](size_t idx) { return buf[idx]; }
+    type &operator[](size_t idx) const { return buf[idx]; }
     virtual void zeroize() { secureZero(buf, sizeof(buf)); }
-};
-
-class EC_GROUP_auto {
-public:
-    EC_GROUP* group;
+  };
 
-    EC_GROUP_auto(int nid) {
-        group = EC_GROUP_new_by_curve_name(nid);
-    }
-    ~EC_GROUP_auto() {
-        EC_GROUP_free(group);
-    }
-    operator EC_GROUP*() {return group; }
-    operator const EC_GROUP*() const { return group; }
-};
-
-class pthread_lock_auto {
- public:
-  pthread_mutex_t* lock;
-
-  pthread_lock_auto(pthread_mutex_t* mutex) {
-    lock = mutex;
-    isLocked_ = (pthread_mutex_lock(lock) == 0);
-    if (!isLocked_) {
-      abort();
+  class pthread_lock_auto
```

Comment:
I believe that this is entirely old and dead code which should have been removed as part of removing our custom DRBG. Though not technically part of this PR. I'll remove it all.

Reply:
Thanks, no need to add it to this PR we can do a follow up.  It just stood out to me when I looked at it.

Reply:
I'm stripping it out because it is pure deletion. Normally I don't like pulling things in but when it is just removing dead code, I'm willing to slip it in a bit more aggressively.